### ISSUE TITLE
CLI doesn't complain when given unknown deployments, pods, or paths

### ISSUE THREAD

**Issue by wmorgan** [runconduit/conduit#49]
For `tap` and `stat`, and maybe other commands, specifying a non-existing deployment or pod  doesn't result in any kind of error message.

E.g. `stat`:
```
william:~/devel/conduit$ conduit stat deploy default/hello
NAME            REQUEST_RATE   SUCCESS_RATE   P50_LATENCY   P99_LATENCY
default/hello         2.5rps         96.00%         352ms         875ms
william:~/devel/conduit$ conduit stat deploy default/hello2
NAME   REQUEST_RATE   SUCCESS_RATE   P50_LATENCY   P99_LATENCY
```

and `tap`:
```
william:~/devel/conduit$ conduit tap deploy default/hello
[172.17.0.3:53280 -> 172.17.0.11:80]
HTTP Request
Stream ID: (0, 263)
HTTP POST world.default.svc.cluster.local/helloworld.World/Greeting
...
^C
william:~/devel/conduit$ conduit tap deploy hello
^C
```

This isn't too bad on its surface, but a) it means that e.g. forgetting to put the namespace for `tap deploy` means that the command just hangs forever (this is an easy mistake, and we had a user confused about this in #conduit today), and b) it doesn't match how `kubectl` works, e.g.

```
william:~/devel/conduit$ kubectl get -n conduit deploy/prometheus
NAME         DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
prometheus   1         1         1            1           7d
william:~/devel/conduit$ kubectl get -n conduit deploy/prometheus2
Error from server (NotFound): deployments.extensions "prometheus2" not found
```

I think these commands should return an error if the specifying pod, deployment, or path (maybe?) is not found.

**Comment by siggy** [comment]
This issue is two-thirds resolved.

## tap

`conduit tap` continues to exhibit this behavior.

```bash
$ conduit tap deploy foo
...
```

## stat

`stat` was fixed in #760:

```bash
$ conduit stat deploy emoji
Error: error calling stat with request: deployment.apps "emoji" not found
Usage:
  conduit stat [flags] RESOURCETYPE [RESOURCENAME]
```

## get

`conduit get` no longer supports resources as input:

```bash
$ conduit get emojivoto/web-b8f8786b8-49d9j
Error: invalid resource type emojivoto/web-b8f8786b8-49d9j, only pods are allowed as resource types
Usage:
  conduit get [flags] pods
```

**Comment by siggy** [comment]
Resolved in #827.
